components/service/fretboard/src/main/java/mozilla/components/service/fretboard/ExperimentEvaluator.kt

```diff
@@ -30,19 +30,23 @@ internal class ExperimentEvaluator(private val regionProvider: RegionProvider? =
 
     private fun matches(context: Context, experiment: Experiment): Boolean {
         if (experiment.match != null) {
-            val region = regionProvider?.getRegion()
+            val region = valuesProvider.getRegion(context)
             val matchesRegion = !(region != null &&
                                         experiment.match.regions != null &&
                                         experiment.match.regions.isNotEmpty() &&
                                         experiment.match.regions.none { it == region })
-            val appVersion = context.packageManager.getPackageInfo(context.packageName, 0).versionName
+            val releaseChannel = valuesProvider.getReleaseChannel(context)
+            val matchesReleaseChannel = !(releaseChannel != null &&
+                                                experiment.match.releaseChannel != null &&
+                                                releaseChannel != experiment.match.releaseChannel)
```

Comment:
With [De Morgan's Law ](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) you can use the complement which may be easier to read (doesn't need the double negation):

```Kotlin
 val matchesReleaseChannel = releaseChannel == null ||
    experiment.match.releaseChannel == null ||
    releaseChannel == experiment.match.releaseChannel
```